Tasks/AzureRmWebAppDeploymentV4/Tests/DeploymentFactoryL0Tests.ts

```diff
@@ -0,0 +1,105 @@
+import tl = require('vsts-task-lib');
+import { DeploymentFactory } from '../deploymentProvider/DeploymentFactory';
+import { BuiltInLinuxWebAppDeploymentProvider } from '../deploymentProvider/BuiltInLinuxWebAppDeploymentProvider'
+import { IWebAppDeploymentProvider } from '../deploymentProvider/IWebAppDeploymentProvider';
+import { TaskParametersUtility, TaskParameters, DeploymentType } from '../operations/TaskParameters';
+import { stringify } from 'querystring';
+import { PackageType } from 'webdeployment-common/packageUtility';
+
+export class DeploymentFactoryL0Tests  {
+
+    public static async startDeploymentFactoryL0Tests() {
+        await DeploymentFactoryL0Tests.testForLinuxWebDeploymentProvider();
+        await DeploymentFactoryL0Tests.testForWindowsWebAppRunFromZipProvider();
```

Comment:
publishprofileprovider, container provider
Also can we cover providers for _getUserSelectedDeploymentProviderForWindow and _getWindowsDeploymentProviderForZipAndFolderPackageType
As discussed offline just trying to maintain all if conditions are present even if in future someone refactors.

Reply:
Done. Added in this commit 55eb84f00332369924f7fbb5891bbc0380ea5d22